android/src/main/java/com/github/wuxudong/rncharts/charts/ChartBaseManager.java

```diff
@@ -308,6 +314,83 @@ public void setMarker(Chart chart, ReadableMap propMap) {
         chart.setMarker(marker);
     }
 
+    @ReactProp(name = "imageMarker")
+    public void setImageMarker(Chart chart, ReadableMap propMap) {
+        if (!BridgeUtils.validate(propMap, ReadableType.Boolean, "enabled") || !propMap.getBoolean("enabled")) {
+            chart.setMarker(null);
+            return;
+        }
+
+        if (!BridgeUtils.validate(propMap, ReadableType.String, "resourceName")) {
+            throw new IllegalArgumentException("expecting resourceName for imageMarker");
+        }
+
+        String resourceName = propMap.getString("resourceName");
+        Float offsetX = 0f;
+        Float offsetY = 0f;
+        Float width = 0f;
+        Float height = 0f;
+        Boolean skipGroupedGoals = false;
+
+        if (BridgeUtils.validate(propMap, ReadableType.Number, "offsetX")) {
+            offsetX = (float)propMap.getDouble("offsetX");
+        }
+        if (BridgeUtils.validate(propMap, ReadableType.Number, "offsetY")) {
+            offsetY = (float)propMap.getDouble("offsetY");
+        }
+        if (BridgeUtils.validate(propMap, ReadableType.Number, "width")) {
+            width = (float)propMap.getDouble("width");
+        }
+        if (BridgeUtils.validate(propMap, ReadableType.Number, "height")) {
+            height = (float)propMap.getDouble("height");
+        }
+        if (BridgeUtils.validate(propMap, ReadableType.Boolean, "skipGroupedGoals")) {
+            skipGroupedGoals = (boolean)propMap.getBoolean("skipGroupedGoals");
+        }
+
+        IMarkerConditionFn skipNonGoal = 
+            skipGroupedGoals ? 
+            // skip drawing halo on highlighted unless it represents a single goal or highlight was set programatically
+            new IMarkerConditionFn() {
+
+                @Override
+                public boolean call(Entry e, Highlight h) {
+                    boolean skip = true;
+                    Map entryData = (e.getData() instanceof Map) ? (Map)e.getData() : new HashMap();
+                    List goalIds = entryData.containsKey("goalIds") ? ((List)entryData.get("goalIds")) : null;
+                    Object highlightMeta = (h instanceof HighlightWithMeta) ? ((HighlightWithMeta) h).getMetaData() : null;
+
+                    if (goalIds != null) {
```

Comment:
wonder if there's a way to support this without needing to rely on switchback logic (e.g. being agnostic to what `goalIds` means, and what `skipGroupedGoals` means?

e.g. maybe instead we add `{:highlightMeta {:autoMark false}}` for the grouped goals, and `{:highlightMeta {:highlightEnabled false}}` for the non-goal data entries.

Reply:
great suggestion. updated!